src/workspacefactory/wfactory_controller.js

```diff
@@ -45,9 +46,28 @@
  * @param {string} previewDiv Name of div to inject preview workspace in.
  * @constructor
  */
-WorkspaceFactoryController = function(toolboxName, toolboxDiv, previewDiv) {
+function WorkspaceFactoryController(toolboxName, toolboxDiv, previewDiv) {
   // Toolbox XML element for the editing workspace.
   this.toolbox = document.getElementById(toolboxName);
+  // Currently displayed toolbox. Upon init, the current toolbox name is
+  // the default toolbox name, which is the empty string. Users are forced
+  // to rename this once they create multiple toolboxes.
+  this.currentToolbox = '';
```

Comment:
Add type annotation to this new member variable.

(And if you don't mind, the other variables below.  But not a requirement.)